sage/testing/src/lib.rs

```diff
@@ -0,0 +1,399 @@
+// Ajuna Node
+// Copyright (C) 2022 BlogaTech AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Helpers to easily create a mocked runtime that is able to host the complete SAGE stack.
+//!
+//! You still need the following boilerplate code in the test suite, which can unfortunately not be
+//! avoided as we need to get some types from there.
+//!
+//! ```rust
+//! use sage_testing::{impl_ajuna_seasons, impl_core_pallets, impl_pallet_sage, impl_test_runtime_sage_api, TestRandomness};
+//! use sp_core::H256;
+//!
+//! // The game logic implemented by the game dev
+//! use example_game::{GameAssetId, GameAsset, GameTransition, TransitionConfig};
+//!
+//! type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<TestRuntime>;
+//!
+//! frame_support::construct_runtime!(
+//!    pub enum TestRuntime
+//!      {
+//!         System: frame_system,
+//!         Timestamp: pallet_timestamp,
+//!         Balances: pallet_balances,
+//!         Assets: pallet_assets,
+//!         AjunaSeasons: pallet_ajuna_seasons,
+//!         Sage: pallet_sage,
+//!      }
+//! );
+//!
+//! pub struct SageEngine;
+//! // Use `TestRandomness` for predictable output.
+//! // Otherwise, we can instantiate the `pallet_insecure_randomness_collective_flip`
+//! // in the runtime.
+//! pub type Randomness = TestRandomness<TestRuntime>;
+//!
+//! impl_core_pallets!(TestRuntime, System);
+//! impl_ajuna_seasons!(TestRuntime, GameAssetId, Sage);
+//! impl_pallet_sage!(TestRuntime, GameTransition, GameAssetId, GameAsset, AjunaSeasons, Balances, Assets);
+//! impl_test_runtime_sage_api!(
+//!   SageEngine,
+//!   TestRuntime,
+//!   Sage,
+//!   AjunaSeasons,
+//!   GameAssetId,
+//!   GameAsset,
+//!   TransitionConfig,
+//!   Randomness,
+//!   H256
+//! );
+//! ```
```

Comment:
This piece of code, and defining the TestExternalities should be the only thing needed to get a full test setup.

I am thinking about how to get a generic `TestExternalities` setup, but I think this should actually be very straight forward with generics.